lib/ansible/modules/cloud/amazon/aws_organization_units.py

```diff
@@ -255,27 +258,21 @@ def main():
         result['state'] = 'present'
         result['ou'] = ou
 
-    if module.check_mode:
-        if ou_state == 'absent':
-            if ou is None:
-                result['changed'] = False
-            else:
-                result['changed'] = True
-        elif ou_state == 'present':
-            if ou is None:
+    if ou_state == 'absent':
+        if ou is None:
+            result['changed'] = False
+        else:
+            if module.check_mode:
                 result['changed'] = True
```

Comment:
Just to reduce duplication, you could switch the 'else' to a ``if not module.check_mode:`` and leave the ``result['changed'] = True`` outside the if statement.  the result won't be returned if an exception is thrown while making the change.   (and the same below)

Reply:
Realised I've explained poorly:

```
if ou is None:
  result['changed'] = False
else:
  result['changed'] = True
  if not module.check_mode:
    client.delete_ou(ou['Id'])
    result['state'] = 'absent'
```

Reply:
This looks much better, thanks